### PR TITLE
fix: clone value instead taking ownership for dedupe

### LINKED ISSUE #3005: bug: dedupe is failing
when you run the following schema with k6 for benchmarking, some portion of requests fail with following error.
error:  
```json
{"data":null,"errors":[{"message":"Deduplication failed"}]}
```

```graphql
schema
  @server(port: 8000)
   @upstream(baseURL: "http://jsonplaceholder.typicode.com"){
  query: Query
}

type Query {
  posts: [Post] @http(path: "/posts", dedupe: false)
}

type Post {
  id: Int!
  title: String!
}
```

k9 setup
```js
import http from 'k6/http';
import { check } from 'k6';

export const options = {
  stages: [
    { duration: '30s', target: 50 },
  ],
};

export default function () {
  let query = `
  query GetPosts {
    posts {
      id
    }
}`;

  let headers = {
    'Content-Type': 'application/json',
    'GraphQL-Client-Name': 'k6',
    'GraphQL-Client-Version': '0.0.1',
  };

  let res = http.post('http://localhost:8000/graphql', JSON.stringify({ query: query, operationName: 'GetPosts' }), {
    headers: headers,
  });
  check(res, {
    'Status 200': (r) => r.status === 200 && r.body.includes('errors') === false,
  });
}
```

### PR DESCRIPTION
**Pull request by laststylebender14**
**Summary:**  
_Briefly describe the changes made in this PR._

**Issue Reference(s):**  
Fixes #3005

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [x] I have performed a self-review of my code.
- [x] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: tests/core/http.rs [hunk 1]
```diff
@@ -55,6 +55,9 @@ impl Http {
 #[async_trait::async_trait]
 impl HttpIO for Http {
     async fn execute(&self, req: reqwest::Request) -> anyhow::Result<Response<Bytes>> {
+        // configure wait to simulate the actual real world request.
+        let _ = tokio::time::sleep(tokio::time::Duration::from_millis(10)).await;
+
```
Comment (meskill) [inline tests/core/http.rs:60]: let's better do the test changes as a separate pr I'm not sure about them right now

File: src/core/jit/graphql_executor.rs [hunk 2]
```diff
@@ -68,10 +68,13 @@ impl JITExecutor {
...
-        let val = out.unwrap_or_default();
-        Arc::into_inner(val).unwrap_or_else(|| {
-            Response::from_errors(vec![ServerError::new("Deduplication failed", None)])
-        })
+
+        // Using Arc it's easy to fool the dedupe, but it can be problematic if the
+        // underlying value isn't cloneable. To ensure deduplication works
+        // correctly and to maintain consistency, we clone the response before
+        // sending it out. This approach guarantees that response value is
+        // available to all deduplicated requests.
```
Comment (meskill) [inline src/core/jit/graphql_executor.rs:76]: better not to store data as `Arc` in that case, please, drop it from dedupe `dedupe_operation_handler`

**Review by meskill — CHANGES_REQUESTED** [review]

File: src/core/jit/graphql_executor.rs (hunk 2 above)
Comment (meskill) [reply src/core/jit/graphql_executor.rs:76]: create a simple wrapper with clone implementation for `Response` to bypass the type restrictions.

Comment (laststylebender14) [reply src/core/jit/graphql_executor.rs]: done, using `Lift` for cloning.

File: tests/core/http.rs (hunk 1 above)
Comment (laststylebender14) [reply tests/core/http.rs]: done, dropped test related changes, will raise separate issue for this.

**Review by meskill — APPROVED** [review]

**Pull request merged**